### DISCUSSION TITLE
How to DoT/DoH with Traefik as reverse proxy

### DISCUSSION THREAD

**Discussion by LucEast** [AdguardTeam/AdGuardHome#5292] [Q&A]
Hi,

i want to use AdGuard on a cloud vHost with Docker. I use Traefik as my reverse proxy and for automatic certificate renewal.
Now I want to set up DoT/DoH.
In the encryption settings I need to specify the cert path and the private key.
Is there any way to use the cert from Traefik?

**Comment by ammnt** [comment]
> Hi,
> 
> i want to use AdGuard on a cloud vHost with Docker. I use Traefik as my reverse proxy and for automatic certificate renewal. Now I want to set up DoT/DoH. In the encryption settings I need to specify the cert path and the private key. Is there any way to use the cert from Traefik?

You need to setup HTTPS only for Traefik reverse proxy and no need to enable HTTPS in AdGuard Home settings. DoH can work over HTTP without excessive encryption (80 port). Just try to enable this option:

```
tls:
  allow_unencrypted_doh: true
```
More information here:
https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#disable-doh-encryption-on-adguard-home

**Reply by trmdi** [reply]
What about DoT? My similar question: https://github.com/AdguardTeam/AdGuardHome/discussions/5867
